BUG: Catch error if array-priority is not float compatible

seberg opened the pull request:
This is a bit cleaner than before.  This shouldn't lead to serious
issues, normally just a SystemError, because the error is not cleared.
But maybe it is a bit cleaner.

EDIT: Moved from gh-21217  something went very wrong there after a force-push (that had failed on the github side the first time)

~EDIT2: Well, I guess this will take a bit longer ;).  Now CI is acting up, seems github just has some issues right now...~

seberg committed:
BUG: Catch error if array-priority is not float compatible

This is a bit cleaner than before.  This shouldn't lead to serious
issues, normally just a SystemError, because the error is not cleared.
But maybe it is a bit cleaner.

mattip commented on numpy/core/src/multiarray/multiarraymodule.c:
```diff
@@ -129,12 +129,18 @@ PyArray_GetPriority(PyObject *obj, double default_)
-            PyErr_Clear(); /* TODO[gh-14801]: propagate crashes during attribute access? */
+            /* TODO[gh-14801]: propagate crashes during attribute access? */
+            PyErr_Clear();
         }
```
Strange, this error condition is still not hit. Maybe the test needs a `__getitem__` that raises?

seberg replied:
ohhh, I did not modify this one, so didn't add a test for it.  Let me just add it, since it fits into the same "theme".

seberg committed:
TST: Add test for `__array_priority__` raising a bad error

As the TODO note states, this should probably just raise the error,
but for now, lets cover the code path and make sure it doesn't do
worse at least.

mattip commented:
Thanks @seberg
